### PR TITLE
Save document title after edit

### PR DESCRIPTION
**Pull request by elpoelma**
This ensures that the title of the document is saved just after having edited it. An additional save of the document is no longer required.

### CODE REVIEW

**Comment by nvdk** [comment]
one consideration is that this doesn't create a new version of the document, which is different from before. open question whether this is what a user expects. (phrased another way: do title changes warrant a new version?)

**Comment by nvdk** [comment]
@elpoelma can you fix the merge conflict?

**Comment by elpoelma** [comment]
> @elpoelma can you fix the merge conflict?

Done, can this be merged?

**Comment by nvdk** [comment]
I'm hesitant because of the above remark :)

**Comment by abeforgit** [comment]
semantically, I'd say yes, title changes warrant a new version
versions are pretty transparent to the user anyway

**Review by nvdk — CHANGES_REQUESTED** [review]
looked into this a bit more and we were already saving on onSubmit as far as I can see, so we should clean this up. because now we are calling the component with 
```js
      <EditorDocumentTitle @title={{@editorDocument.title}} @editActive={{@editorDocument.isNew}} 
        @onChange={{this.updateDocumentTitle}} @onSubmit={{this.saveDocument}}  @onCancel={{this.resetDocument}} 
```
and both onSubmit and onChange save the document ? at the very least this should be cleaned up.

**Comment by nvdk** [comment]
I've reworked this a bit + fixed some other stuff I ran into, needs a proper review @elpoelma :)

File: app/components/app-chrome.js [hunk 1]
```diff
@@ -19,7 +19,7 @@ export default class AppChromeComponent extends Component {
   }
 
   @action
-  updateDocumentTitle(title) {
-    this.args.editorDocument.title = title;
+  async resetDocument() {
```
Comment (nvdk) [inline app/components/app-chrome.js:22]: woops I think this bit is no longer relevant and can go

**Comment by elpoelma** [comment]
There was some small issue in the `cancel` method, otherwise now it seems to work well. TItle edits also add a new version correctly now which shows up in the revisions tab.

**Comment by elpoelma** [comment]
I'll update the reglementaire bijlage frontend in order to have the same workflow tomorrow.

File: app/components/editor-document-title.js [hunk 2]
```diff
@@ -5,27 +5,43 @@ import { tracked } from '@glimmer/tracking';
...
 
     if (title) {
       this.error = false;
     }
   }
 
+  @action
+  submit(event) {
+    event.preventDefault();
+    console.log(event, this.title);
```
Comment (nvdk) [inline app/components/editor-document-title.js:31]:
```suggestion
```

**Pull request merged**